app/models/participant_profile/ecf.rb

```diff
@@ -46,6 +48,12 @@ def fundable?
   def policy_class
     ParticipantProfile::ECFPolicy
   end
+
+private
+
+  def update_analytics
+    Analytics::UpsertECFParticipantProfileJob.perform_later(participant_profile: self) if saved_change_to_training_status?
+  end
```

Comment:
Why has this moved to only `ParticipantProfile::ECF` this would stream anymore NPQ participant changes.

Reply:
I found that this analytics job was erroring when running for NPQ participants during the tests when the participants were being created. Which makes sense as `ECFValidationService` doesn't sound like it should be running for NPQ participants. I don't have a good explanation as to why we don't see lots of errors in sentry on production for this, maybe because we aren't changing the training status column?

I think it was breaking on this line https://github.com/DFE-Digital/early-careers-framework/blob/CPDRP-1033-sidekiq-migration/app/services/analytics/ecf_validation_service.rb#L47 because NPQ profiles don't have a `ecf_participant_eligibility` relation